include/flexiv/Robot.hpp

```diff
@@ -348,191 +374,315 @@ class Robot
-     * RT/NRT_CARTESIAN_MOTION_FORCE_TCP.
-     * @throw InputException if input is invalid.
-     * @throw LogicException if robot is not in the correct operation mode.
-     * @throw ExecutionException if error occurred during execution.
      */
-    void setNullSpacePosture(
-        const std::vector<double>& preferredPositions = {0, 0, 0, 0, 0, 0, 0});
+    void setNullSpacePosture(const std::vector<double>& preferredPositions);
 
-    //=============================== IO CONTROL ===============================
     /**
-     * @brief Set digital output on the control box.
-     * @param[in] portNumber Port to set value to [0 ~ 15].
-     * @param[in] value True: set high, false: set low.
-     * @throw ExecutionException if error occurred during execution.
-     * @throw InputException if input is invalid.
+     * @brief [Non-blocking] Reset preferred joint positions to the robot's home
+     * posture.
+     * @note Applicable control modes: RT/NRT_CARTESIAN_MOTION_FORCE.
      */
-    void writeDigitalOutput(unsigned int portNumber, bool value);
+    void resetNullSpacePosture(void);
 
     /**
-     * @brief Read digital input on the control box.
-     * @param[in] portNumber Port to read value from [0 ~ 15].
-     * @return True: port high, false: port low.
-     * @throw CommException if there's no response from server.
-     * @throw ExecutionException if error occurred during execution.
+     * @brief [Blocking] Set force-controlled Cartesian axis(s) for the
+     * Cartesian motion-force control modes. The axis(s) not enabled for force
+     * control will be motion controlled. This function can only be called when
+     * the robot is in IDLE mode.
+     * @param[in] enabledAxis Flags to enable/disable force control for certain
+     * Cartesian axis(s) in the force control reference frame (configured by
+     * setForceControlFrame()). The corresponding order is \f$ [X, Y, Z, Rx, Ry,
+     * Rz] \f$. By default, force control is disabled for all Cartesian axes.
      * @throw InputException if input is invalid.
-     */
-    bool readDigitalInput(unsigned int portNumber);
+     * @throw LogicException if robot is not in the correct control mode.
+     * @throw ExecutionException if failed to execute the request.
+     * @note Applicable control modes: IDLE.
+     * @warning This setting will reset to all axes disabled upon disconnection.
+     * @warning This function blocks until the request is successfully delivered
+     * to the robot.
+     */
+    void setForceControlAxis(const std::vector<bool>& enabledAxis);
+
+    /**
+     * @brief [Blocking] Set force control reference frame for the Cartesian
+     * motion-force control modes. This function can only be called when the
+     * robot is in IDLE mode.
+     * @param[in] referenceFrame The reference frame to use for force control.
+     * Options are: "TCP" and "BASE". The target wrench and force control axis
```

Comment:
To clarify, in v0.9 the reference frame name options are `TCP` and `BASE`, but in v1.2 the reference frame options are `TCP` and `WORLD`. Suggest syncing this to the same name to avoid confusion.

Reply:
Base and world are two different frames and the doc here accurately states the fact that v0.9 and v1.2 use different global coordinates.